compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1933,22 +1934,37 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
     }
 
     /// Checks if the `#[align]` attributes on `item` are valid.
-    fn check_align(&self, span: Span, target: Target, align: Align, repr_span: Span) {
+    fn check_align(
+        &self,
+        span: Span,
+        hir_id: HirId,
+        target: Target,
+        align: Align,
+        attr_span: Span,
+    ) {
         match target {
             Target::Fn | Target::Method(_) | Target::ForeignFn => {}
+            Target::Field => {
+                self.tcx.emit_node_span_lint(
+                    UNUSED_ATTRIBUTES,
+                    hir_id,
+                    attr_span,
+                    AlignOnFields { span },
+                );
+            }
```

Comment:
Question: do we already have test coverage for this case?

Reply:
We have about 20 of these cases and to the best of my knowledge no test coverage for any :ferris clueless:

Reply:
Don't really have time to atm, am on holiday. Just knew how to fix this issue well

Reply:
feel free to assign me or @JonathanBrouwer (who I think was interested in this sort of thing)